0001-typedb/01-start/04-iam-schema.md

```diff
@@ -0,0 +1,302 @@
+
+- `action-name`
+- `object-type`
+- `parent-company`
+
+Additionally, `action` can play a role in multiple relations:
+
+- `access` relation as role `valid-action`
+- `company-membership` as role `company-member`
+- `segregation-policy` as role `segregated-action`
+- `set-membership` as role `set-member`
+
+Finally, `action` has two subtypes, which are not abstract, so we can create instances of those subtypes:
+
+- `operation` — a single action type that can be performed with an object
+- `operation-set` — a set of actions that can be performed with an object
+
+Both subtypes inherit all the attribute and relation types defined in the action type.
+
+## Membership subtypes
+
+The `membership` is a relation type that has multiple subtypes for different kinds of relations, regarding membership 
+in groups:
+
+- `membership`:
+  - `collection-membership` — combines objects in resource-collections
+  - `group-membership` — combines subjects in user-groups
+  - `set-membership` — combines actions in operation-sets
+
+## Ownership subtypes
+
+The `ownership` is a relation type that has multiple subtypes for different kinds of relations, regarding ownership 
+groups:
+
+- `ownership`:
+  - `object-ownership` — assigns an owner of the subject type for an object
+  - `group-ownership` — assigns an owner of the subject type for a user-group
+
+## Segregation policy
+
+Relation type that adds information on [duty segregation](https://en.wikipedia.org/wiki/Separation_of_duties) policies. 
+It has a `name` attribute and a single role `segregated-action`. Usually, multiple instances of this role in a single 
+relation mean these actions can’t be performed by one person.
+
+## Rules
+
+There are multiple rules in the schema that can be used in different situations and queries.
+
+1. `attribute-parent-company`
+2. `automatic-member-collection`
+3. `automatic-root-collection`
+4. `transitive-membership`
+5. `transitive-object-access`
+6. `transitive-action-access`
+7. `transitive-subject-permission`
+8. `transitive-object-permission`
+9. `transitive-action-permission`
+10. `automatic-segregation-violation`
+11. `automatic-permission-invalidity`
+12. `automatic-permission-validity`
+13. `add-view-permission`
+
+### add-view-permission rule
+
+This simple rule illustrates basic inference. More information on rules can be found in the 
+[schema](../02-dev/02-schema.md#rules) documentation page.
+
+<!-- test-ignore -->
+```typeql
+define
+
+rule add-view-permission:
+when {
+    $modify isa action, has action-name "modify_file";
+    $view isa action, has action-name "view_file";
+    $ac_modify (accessed-object: $obj, valid-action: $modify) isa access;
+    $ac_view (accessed-object: $obj, valid-action: $view) isa access;
+    (permitted-subject: $subj, permitted-access: $ac_modify) isa permission;
+} then {
+    (permitted-subject: $subj, permitted-access: $ac_view) isa permission;
+};
+```
+
+The `when` clause defines the following conditions:
```

Comment:
The numbered list looks like a paraphrasing of the rule and doesn't particularly help me understand the intent.
I prefer the sentences we usually use to describe the intent of the rule:

> If $view the "view_file" action, and $modify is the "modify-file" action:  
Whenever a subject `$subj` has access to has permissions to the access (`$modify` on `$obj`), then `$subj` also has permission to the access ($view on $obj), provided both $view and $modify are valid access modes on $obj.

Reply:
We have a whole dedicated section called Explanation down there, that does that in detail. The numbered list is practically explaining every line, yes. We need both here, because it's not just an example.

Reply:
Yes. They usually use `unify` rather than `assign`. But that's not a common term outside of it.

Reply:
> Yes. They usually use `unify` rather than `assign`. But that's not a common term outside of it.

That would be confusing as F*ork* to use `unify` instead of assign. I suggest we stick to assign for now. I double-checked and it looks like all instances of assign are pretty much straightforward. Shouldn't cause any confusion as it is right now IMHO.

BTW, All we do here with variables is just assign them in the context of this particular query. Nothing persistent.